### DISCUSSION TITLE
Feature request: Ability to create multiple windows

### DISCUSSION THREAD

**Discussion by dvcrn** [notaapp/talk#299] [Feature Requests]
I'd like to have multiple workspaces open at the same time in different windows, but currently it doesn't look like that's possible 

Would be cool to have this added at some point

**Comment by astoilkov** [comment]
Can you describe your use cases? I am asking because I and @erusev don't feel the need for multiple windows and we are not sure what the use cases are.

At first, we were considering multiple windows because we wanted to switch between workspaces fast. However, after implementing the Swift Workspace... dialog that need went away as this workflow is good at switching workspaces quickly.

Let the voting begin.

**Reply by dvcrn** [reply]
Mostly for having multiple things open at the same time. For me specifically, I have 2 main vaults, one work specific one private. Both serve a different purpose and it would be nice to be able to have them open side-by-side, so I can tab between them with CMD+`. I did something similar with Obsidian, and currently solve this issue by having one obsidian window and one nota window

**Reply by astoilkov** [reply]
Got it. Yes, this use case makes a lot of sense. Thanks for the clarification.

**Reply by petrbouchal** [reply]
For me, this would be for having two different notes on two screens or side by side, even in one workspace.

**Reply by webwesen** [reply]
one (long) note could be open in 2 windows (`right` for writing, `left` for reference or some such)

**Comment by Strajk** [comment]
I'm so dependant on Nota that I took some time to dive into how Electron & Mac app works, here's a workaround to open multiple instances:

* Install asar, required to extract the Electron app.asar archive.
  * `npm install -g asar`
* Extract the app.asar archive
  * `asar extract /Applications/Nota.app/Contents/Resources/app.asar /Applications/Nota.app/Contents/Resources/app`
* Rename the app.asar, so Electron uses the extracted app folder instead
  * `mv /Applications/Nota.app/Contents/Resources/app.asar /Applications/Nota.app/Contents/Resources/app.asar.bak`
* Remove the `requestSingleInstanceLock` lock from main.js files
  * Manually change `g.app.requestSingleInstanceLock() ? rO() : g.app.quit();` to `rO();`
  * Or run 
    ```
	sed -i '' 's/g.app.requestSingleInstanceLock() ? rO() : g.app.quit();/rO();/g' /Applications/Nota.app/Contents/Resources/app/main.js
    ```
* Open the app using -n flag to open a new instance
  * `open -n -a /Applications/Nota.app`

![Screenshot-2024-09-14T09 39 43](https://github.com/user-attachments/assets/a80375b8-3526-4898-9681-28ebcb296171)
